src/pages/groupComparison/vennDiagram/VennUtils.ts

```diff
@@ -0,0 +1,391 @@
+import _ from "lodash";
+import {nelderMead} from 'fmin';
+import {getDeterministicRandomNumber} from "../../../shared/components/plots/PlotUtils";
+
+type Area = {size:number, sets:string[], preciseSize: number};
+type SetRectangles = {[setId:string]:Rectangle};
+type Rectangle = {x:number, y:number, xLength:number, yLength:number};// bottom-left aligned
+const VennJs = require("venn.js");
+
+function getMargin(x:number, y:number, setRectangles:SetRectangles) {
```

Comment:
would be nice to have comment summarizing the logic here.  like, why do we choose the min of these various calculations?